### ISSUE TITLE
Make DownloadMaxTries configurable

### ISSUE THREAD

**Issue by beatcracker** [Otiel/BandcampDownloader#13]
Default `10` in the release version is too low. Out of curiosity, I've patched it with Reflexil to `int32.MaxValue` and found that it needed ~`80` retries to complete download in my case. Plus, could we have infinite retries?

Oh, and thanks! This tool is very convenient for those who don't want to keep Bandcamp open in the browser all the time.

**Comment by Otiel** [comment]
I could add that as an option. I don't feel confident setting infinite retries by default because that would induce a lot of network requests for nothing if for some reason you let the program work and Bandcamp is blocking you.

**Comment by beatcracker** [comment]
Thanks!

P.S> Something like [exponential backoff](https://en.wikipedia.org/wiki/Exponential_backoff) could be used for infinite retires, so we don't flood Bandcamp with requests.

**Comment by Otiel** [comment]
The thing is, I don't know how the Bandcamp anti-spam filter works. I'll have to do some tests in order to see if waiting between retries would be enough.

**Comment by beatcracker** [comment]
If you could provide me with test case (request log or something), I probably could make simple PowerShell script that will d/l files with configurable delays. This'll make it easier to run tests with reproducible results.

**Comment by ajsnyde** [comment]
@Otiel Mind if I give this feature a go? I'd use a spinner for retries, with a combobox for exponential backoff strategy. Alternatively, I can create a little option bar/window on the top rail if needed.

**Comment by Otiel** [comment]
@ajsnyde If you can work something out, be my guest.

**Comment by ajsnyde** [comment]
I've created a partial workaround here:
https://github.com/Otiel/BandcampDownloader/pull/14/commits/25bb3e9e647048096fb12330dfffc64310b99d16
I'm plan to add the retry strategy soon.

**Comment by ajsnyde** [comment]
See comments and changes here: https://github.com/ajsnyde/BandcampDownloader/commit/73bc604100166ede8a231df7ab6df69ee9487984

<blockquote>
"Added retry cooldown settings. DownloadRetryCooldown can be set, which puts the application to sleep after each failed attempt to download (in seconds). DownloadRetryExponential Can be set to increase the cooldown with each successive failed attempt. Default values of 0.0 and 1.0 should leave the application running as it does without this commit. After a little experimentation, I found that using 0.2 seconds and 2.0 exponential factor, and 7 DownloaddMaxTries prevents all failures. I am unable to "spam out" the service and not get my album/songs. All of these settings must be set using the json settings file.
 master"</blockquote>

I've added two more properties to the settings file. As a quick reminder, you will need to re-save your settings file and then edit the values for these changes to take effect.
